messagebus/src/main/java/com/yahoo/messagebus/network/NetworkMultiplexer.java

```diff
@@ -111,19 +112,21 @@ public void detach(NetworkOwner owner) {
         if ( ! owners.remove(owner))
             throw new IllegalArgumentException(owner + " not attached to this");
 
-        if ( ! shared && owners.isEmpty())
-            net.shutdown();
+        countDown();
     }
 
     public void destroy() {
-        if ( ! shared)
-            throw new UnsupportedOperationException("Destroy called on a dedicated multiplexer; " +
-                                                    "this automatically shuts down when detached from");
+        if (destructible.getAndSet(true))
+            throw new IllegalStateException("Destroy called on a dedicated multiplexer--" +
+                                            "this automatically shuts down when detached from--or " +
+                                            "called multiple times on a shared multiplexer");
 
-        if ( ! owners.isEmpty())
-            log.warning("NetworkMultiplexer destroyed before all owners detached: " + this);
+        countDown();
+    }
 
-        net.shutdown();
+    private void countDown() {
```

Comment:
Without knowing much about messagebus, this method does not seem to be counting down anything, so perhaps e.g. 'shutdownIfOwnerless' would be more descriptive?

Reply:
Yes, it would. Used a counter at first, but that wasn't expressive enough. PTAL.